tests/components/bsblan/test_diagnostics.py

```diff
@@ -0,0 +1,127 @@
+"""Tests for the diagnostics data provided by the Elgato integration."""
+from aiohttp import ClientSession
+
+from homeassistant.core import HomeAssistant
+
+from tests.common import MockConfigEntry
+from tests.components.diagnostics import get_diagnostics_for_config_entry
+
+
+async def test_diagnostics(
+    hass: HomeAssistant,
+    hass_client: ClientSession,
+    init_integration: MockConfigEntry,
+):
+    """Test diagnostics."""
+    assert await get_diagnostics_for_config_entry(
+        hass, hass_client, init_integration
+    ) == {
+        "info": {
```

Comment:
Couldn't these three dicts (`info`, `device`, `state`) be read from the fixture JSON files instead of repeating them here?

Sure, the data exposed by diagnostics is a subset of the fixtures, but perhaps this test could compare the extant keys of each dict only...

Reply:
Good thinking. I guess that would do for testing? Not sure as I am not an expert on testing. But other integrations are doing the same as here. (see my previous copy mistake) So I guess this is alright.

Reply:
Ok I took a look and saw the error. Added a fixture for diagnostics